string_utils/fms_string_utils.F90

```diff
@@ -112,11 +114,20 @@ subroutine c_free(ptr) bind(c,name="free")
   module procedure cpointer_fortran_conversion
 end interface
 
-!> Converts a number to a string
+!> Converts a number or a Boolean value to a string
 !> @ingroup fms_mod
 interface string
```

Comment:
I'm starting to think this could be a single routine that uses `class(*)`.  This seems so over complicated for such a simple routine.

Reply:
`class(*)` version implemented in 9aec801e6d325453b9691ac50d0d86aa2ec2ad02. Although to be honest, I'm not sure I see the benefit of `class(*)` compared to the version based on the interface. `stringify` still requires the `r4`/`r8` templating machinery to exist (unless we move `stringify` to a different module). With the `class(*)` version, unless there's some magic trick that I'm not aware of, the code block for `real` arguments needs to be repeated for `r4_kind` and `r8_kind`. But both versions ultimately get the job done.

Reply:
The code block is repeated, but it's all in the same file and a future developer won't have to go looking around to find where it is and where the macros are.